Doctrine extensions

petrparolek opened the pull request:
Improvements in Nettrine example
- added examples of nettrine/extensions-atlantic18
- switched from SQLite to MariaDB in Docker

f3l1x commented on nettrine/README.md:
```diff
@@ -2,16 +2,35 @@
 This is example of nettrine/orm
 
-## Installation
+## Installation with Docker:
```
Cim nestacil puvodni setup s PHP dev serverem?

petrparolek replied:
Problém nastal, když jsem měl DB v SQLite, tak spustění příkazu `php bin/console orm:validate` doctrine hlásila, že DB struktura není aktuální. Napadlo mě vylepšít řešení, které tu bylo předtím s MySQL.

f3l1x replied:
V jednoduchosti je sila, nekomplikujme to dockerem prosim.

petrparolek replied:
smažu z readme

f3l1x commented on nettrine/app/Bootstrap.php:
```diff
@@ -24,6 +26,12 @@ public static function boot(): Configurator
 		$configurator
 			->addConfig(__DIR__ . '/config/common.neon');
 
+		$localNeon = __DIR__ . '/config/local.neon';
```
Bud dejme local.neon jako podminku nebo vubec.

petrparolek replied:
jak to myslíš?

petrparolek replied:
vždyť tam je podmínka na řádku 28

f3l1x commented on nettrine/app/Bootstrap.php:
```diff
@@ -1,4 +1,6 @@
-<?php declare(strict_types=1);
+<?php
```
Codestyle

f3l1x commented on nettrine/app/Model/Database/Entity/Book.php:
```diff
@@ -1,4 +1,6 @@
-<?php declare(strict_types=1);
+<?php
```
Codestyle

f3l1x commented on nettrine/app/Model/Database/Entity/Category.php:
```diff
@@ -1,26 +1,40 @@
-<?php declare(strict_types=1);
+<?php
```
Codestyle

f3l1x commented on nettrine/app/Model/Database/Entity/Category.php:
```diff
@@ -1,26 +1,40 @@
+ * @Gedmo\Loggable
  */
-class Category extends Entity
+class Category extends Entity implements Translatable
```
Ja bych byl pro udelat dalsi sadu entit, aby na zacatecnika hnedka nevyskocilo vsechno.

petrparolek replied:
~~OK a co takhle udělat novou entitu i další stránku s příklady pouze s Translatable?~~ Nechal bych to takhle, ostatní entity jsou základní

petrparolek replied:
@f3l1x ping

f3l1x commented on nettrine/app/Model/Database/Fixtures/BooksFixture.php:
```diff
@@ -15,14 +17,50 @@ class BooksFixture implements FixtureInterface, OrderedFixtureInterface
 	{
+		$connection = $manager->getConnection();
+
+		$sql = "ALTER TABLE ext_translations AUTO_INCREMENT = 1;";
```
K cemu je tu tohle?

petrparolek replied:
když budeš znovu nahrávávat fixtures, tak budeš mít hausnumera v ID.

petrparolek replied:
mám to smazat?

f3l1x replied:
Smaz to prosim. Nechme to jednoduche.

petrparolek replied:
smazat jen v readme? Nebo odstanit i soubory okolo dockeru?

f3l1x replied:
I ty soubory prosim, dikes.

f3l1x commented on nettrine/app/config/local_example.neon:
```diff
@@ -0,0 +1,8 @@
+nettrine.dbal:
```
local.neon.dist

petrparolek replied:
ok

f3l1x commented on nettrine/app/Presenters/HomepagePresenter.php:
```diff
@@ -88,7 +127,8 @@ public function processBookForm(Form $form): void
 	{
 		$form = new Form();
-		$form->addText('title', 'Title');
+		$form->addText("titleEN", 'Title in English');
```
Stejne jako u entity, zde bych vytvoril dalsi formulare.

petrparolek replied:
OK a co takhle udělat novou entitu i další stránku s příklady pouze s Translatable?

f3l1x replied:
Idealni. To by se mi libilo.

petrparolek committed:
Improvements in Nettrine example
- added examples of nettrine/extensions-atlantic18
- switched from SQLite to MariaDB in Docker

f3l1x commented on nettrine/README.md:
```diff
@@ -2,16 +2,22 @@
 This is example of nettrine/orm
 
-## Installation
+## Installation with Docker:
```
Porad tu vidim docker

petrparolek replied:
ty ho opravdu chceš úplně smazat? U datagridu máš taky docker

f3l1x replied:
Jop, prosim smazat. Udelame pak centralni docker image pro vsechny ukazky.

petrparolek replied:
ok, smazáno

f3l1x commented on nettrine/app/Bootstrap.php:
```diff
@@ -24,6 +24,12 @@ public static function boot(): Configurator
 			->addConfig(__DIR__ . '/config/common.neon');
 
+		$localNeon = __DIR__ . '/config/local.neon';
+		if (file_exists($localNeon)) {
```
local.neon bude vzdycky potreba, netreba to podminkovat na file_exists

petrparolek replied:
ok, podmínku smažu

f3l1x commented on nettrine/app/config/common.neon:
```diff
@@ -32,13 +34,33 @@ extensions:
 	connection:
-		driver: 'pdo_sqlite'
-		url: sqlite:///%appDir%/../data/database.sqlite
+		driver: mysqli
```
Proc nestaci sqlite?

petrparolek replied:
protože sqlite mi pořád házela nekonzitestnost DB s entitami a do migrací přívalo nesmysly do nekonečna, tak jsem to vrátil k vému mysql

f3l1x commented on nettrine/app/Presenters/DoctrineExtensionsPresenter.php:
```diff
@@ -0,0 +1,102 @@
+		$articleRepository = $this->em->getArticleRepository();
+		$articlesEN = [];
+
+		foreach ($articleRepository->findAll() as $article) {
```
Co vlastne dela toto? Vsiml jsem si, ze tady je en_GB a cs_CZ. Proc je to tu dvakrat?

petrparolek replied:
![obrazek](https://user-images.githubusercontent.com/6066243/89068879-65732980-d372-11ea-8703-571d18a85067.png)

petrparolek replied:
protože dvakrát vypisuju, viz náhled

petrparolek committed:
removed Docker

petrparolek committed:
Improvements in Nettrine example
- improved translatable example
- added contributte translation
- improved templates
- updated composer packages
- minor changes and improvements

petrparolek committed:
Improvements in Nettrine example
- added example of softDeleteable

petrparolek committed:
Improvements in Nettrine example
- added example of timestampable

petrparolek committed:
Improvements in Nettrine example
- added example of sluggable

petrparolek committed:
Improvements in Nettrine example
- typo

petrparolek committed:
Improvements in Nettrine example
- added example of tree
- improved layout

petrparolek committed:
Improvements in Nettrine example
- typo

f3l1x commented on nettrine/app/Model/Database/Fixtures/ArticlesFixture.php:
```diff
@@ -0,0 +1,65 @@
+		$article->setTranslatableLocale('en_GB');
+
+		$manager->persist($article);
+		$manager->flush();;
```
2x strednik

petrparolek committed:
improved code

petrparolek committed:
added forgotten migration

petrparolek committed:
improved code

petrparolek committed:
improved code
- show articles which are not deleted
- added type hints
- improved rendering date and time in template

petrparolek committed:
updated composer libraries

petrparolek committed:
bugfixes

petrparolek committed:
updated README.md

f3l1x commented on nettrine/app/Bootstrap.php:
```diff
@@ -24,6 +24,12 @@ public static function boot(): Configurator
 		$configurator
 			->addConfig(__DIR__ . '/config/common.neon');
 
+		$localNeonFile = __DIR__ . '/config/local.neon';
```
To uz jsme jednou resili.

f3l1x commented on nettrine/app/Model/Database/Basic/Repository/BookRepository.php:
```diff
@@ -0,0 +1,21 @@
+final class BookRepository extends EntityRepository
+{
+
+	public function getById(int $id): ?Book
```
K cemu je tahle metoda?

f3l1x commented on nettrine/app/Presenters/AdvancedPresenter.php:
```diff
@@ -0,0 +1,191 @@
+
+		$articleCategoryRepository = $this->em->getArticleCategoryRepository();
+		$categories = $articleCategoryRepository->findPairs();
+		$form->addSelect('parent', 'messages.article_categories.parent_category', $categories)
```
Nestacilo by jenom `$this->em->getArticleCategoryRepository()->findPairs()` bez prirazovani do promennych?

f3l1x commented on nettrine/app/Presenters/AdvancedPresenter.php:
```diff
@@ -0,0 +1,191 @@
+
+			$this->flashMessage($this->translator->translate('messages.categories.success_delete'));
+		} catch (Exception $e) {
+			$this->flashMessage($e->getMessage());
```
Nechybi tady rollback, kdyz je to v transakci?

f3l1x commented on nettrine/migrations/Version20200712145622.php:
```diff
@@ -0,0 +1,47 @@
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20200712145622 extends AbstractMigration
```
Kdyz je to 1 PR, udelej jenom jednu migraci prosimte.

petrparolek committed:
improved code

petrparolek committed:
improvements and bugfixes
- merged migrations
- fixed book entity

petrparolek committed:
improvements and bugfixes
- improve routes
- added missing type hints

petrparolek committed:
improvements and bugfixes
- added phpstan
- bugfixes
